src/tools/bump-stage0/src/main.rs

```diff
@@ -198,9 +198,12 @@ struct Stage0 {
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 struct Config {
     dist_server: String,
-    artifacts_server: String,
-    artifacts_with_llvm_assertions_server: String,
-    git_merge_commit_email: String,
+    // There are other fields in the configuration, which will be read by src/bootstrap or other
+    // tools consuming stage0.json. To avoid the need to update bump-stage0 every time a new field
+    // is added, we collect all the fields in an untyped Value and serialize them back with the
+    // same order and structure they were deserialized in.
```

Comment:
I am not sure this will just work - https://github.com/serde-rs/json/blob/master/Cargo.toml#L59 might be enabled in our workspace, but I'm not immediately confident it is. I'm not sure optimizing to avoid editing this code is really necessary, and it seems like a good idea to have an explicit list so we don't clobber something in that list.

Reply:
> I am not sure this will just work - https://github.com/serde-rs/json/blob/master/Cargo.toml#L59 might be enabled in our workspace, but I'm not immediately confident it is.

Hmm, [I explicitly enabled `preserve_order` in `bump-stage0`'s `Cargo.toml`](https://github.com/rust-lang/rust/pull/99149/files#diff-2009b1fffd2469e0d850cec264fa55af7b73b7a8162d681ae15b589a78176f9b): it'd be quite problematic if that didn't enable the feature. Or do you mean something else?

> I'm not sure optimizing to avoid editing this code is really necessary, and it seems like a good idea to have an explicit list so we don't clobber something in that list.

I added this commit because I originally forgot to add the new field to `bump-stage0` :sweat_smile:

Reply:
I... completely missed that change to Cargo.toml, not sure how. OK, taking another look it looks like I also managed to miss that we do keep a full list in bootstrap's testing, so that solves that concern too.

However, enabling that feature probably comes at some performance cost, and I think will affect the workspace? At least, it'll affect all bootstrap tools -- which I think might these days include some things that we ship, though not confident on that -- so let's run a try build and perf just in case, and then r=me.